src/app/services/crud.service.ts

```diff
@@ -15,34 +15,32 @@ export class CrudService {
   private basePath = '/car';
 
   constructor(
-    private db: AngularFireDatabase,
+    private db: AngularFirestore,
     private dialog: MatDialog
   ) {}
 
-  
   addCar(data: CarData): void {
-    this.db.list(this.basePath).push(data);
+    this.db.collection(this.basePath).add(data);
   }
 
   updateCar(id: string, data: CarData): Promise<void> {
-    return this.db.object(`${this.basePath}/${id}`).update(data);
+    return this.db.collection(this.basePath).doc(id).update(data);
   }
 
   getCarList(): Observable<CarData[]> {
-    return this.db.list<CarData>(this.basePath).valueChanges();
+    return this.db.collection<CarData>(this.basePath).valueChanges({ idField: 'id' });
   }
 
   getCarById(id: string): Observable<CarData> {
-    return this.db.object<CarData>(`${this.basePath}/${id}`).valueChanges().pipe(
-      map(car => car || {} as CarData) 
+    return this.db.collection<CarData>(this.basePath).doc(id).valueChanges().pipe(
+      map(car => car || {} as CarData)
     );
```

Comment:
Tutaj mapowanie jest niepotrzebne. Tak naprawdę zwracasz ten sam obiekt `car` bez żadnej zmiany, więc wystarczy jak zrobisz `return this.db.collection<CarData>(this.basePath).doc(id).valueChanges()`